docs/ref/csp.txt

```diff
@@ -0,0 +1,296 @@
+.. _content-security-policy:
+
+=============================
+Content Security Policy (CSP)
+=============================
+
+.. versionadded:: 5.2
+
+.. module:: django.middleware.csp
+   :synopsis: Middleware for Content Security Policy headers
+
+Content Security Policy (CSP) is a web security standard that helps prevent
+attacks by controlling the sources from which content can be loaded. CSP
+enforcement plays an important role in a comprehensive :doc:`security strategy
+</topics/security>`.
+
+.. _using-csp:
+
+Using CSP
+=========
+
+To enable CSP in your Django project:
+
+1. Add the CSP middleware to your ``MIDDLEWARE`` setting::
+
+      MIDDLEWARE = [
+          # ...
+          "django.middleware.csp.ContentSecurityPolicyMiddleware",
+          # ...
+      ]
+
+2. Configure your CSP policies in your ``settings.py`` using either
+   ``SECURE_CSP`` or ``SECURE_CSP_REPORT_ONLY``::
+
+      from django.middleware import csp
+
+      SECURE_CSP = {
+          "DIRECTIVES": {
+              "default-src": [csp.SELF],
+              "script-src": [csp.SELF],
+              "style-src": [csp.SELF],
+              # Add more directives as needed
+          }
+      }
+
+      # Or for report-only mode:
+      SECURE_CSP_REPORT_ONLY = {
+          "DIRECTIVES": {
+              "default-src": [csp.SELF],
+              "script-src": [csp.SELF],
+              "style-src": [csp.SELF],
+              # Add more directives as needed
+          }
+      }
+
+3. To use CSP nonces for inline scripts or styles, include ``csp.NONCE`` in your
```

Comment:
Can we have a reference to a resource on what a "CSP nonce" even is? The documentation starts out with a nice and friendly welcoming to CSP, but this one stands out as something that's very un-defined.

Perhaps link to the subsequent "Nonce Usage" chapter and reference/introduce it there?

Reply:
Yes, this point would seem better as part of the section below. 🤔

Reply:
In the latest commit I restructure the top level sections to flow a bit better, and moved this example into the section on nonces, as well as added some explanation on what nonces are and why one would use them.